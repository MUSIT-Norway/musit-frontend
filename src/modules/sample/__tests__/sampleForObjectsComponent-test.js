import React from 'react';
import SamplesForObjectComponent from '../SamplesForObjectComponent';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

describe('SamplesForObject', () => {

  it('should display correctly', () => {


    const data = {
      museumObj: 'MUS-123',
      subNo: 'A1',
      term: 'Carex',
      data: [
        {
          sampleType: 'Frø',
          sampleSubType: 'Rogn',
          status: 2,
          createdDate: '1991-11-11'
        },
        {
          sampleType: 'Vev',
          sampleSubType: 'Blod',
          status: 1,
          createdDate: '1991-11-12'},
        {
          sampleType: 'Tekstil',
          sampleSubType: 'Bomull',
          status: 2,
          createdDate: '2011-11-11'}
      ]
    }

    const wrapper = shallow(<SamplesForObjectComponent
      sampleStore={data}
    />);

    expect(shallowToJson(wrapper)).toMatchSnapshot();


  });
});
