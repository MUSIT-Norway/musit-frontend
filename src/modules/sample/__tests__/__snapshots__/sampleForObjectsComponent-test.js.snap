// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SamplesForObject should display correctly 1`] = `
<div>
  <h1>
    Prøver for objekt
  </h1>
  <br />
  <Well
    bsClass="well"
  >
    <b>
      Objektinfo
    </b>
    <br />
    <br />
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        md={2}
      >
        <b>
          MusNo:
        </b>
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        md={2}
      >
        <b>
          SubNo: 
        </b>
        A1
      </Col>
      <Col
        bsClass="col"
        componentClass="div"
        md={3}
      >
        <b>
          Term/Species: 
        </b>
        Carex
      </Col>
    </Row>
  </Well>
  <br />
  <Table
    className="table"
    columns={
      Array [
        Object {
          "key": "id",
          "label": "Sample ID",
        },
        Object {
          "key": "doneDate",
          "label": "Utført Dato",
        },
        Object {
          "key": "sampleType",
          "label": "Prøvetype",
        },
        Object {
          "key": "sampleSubType",
          "label": "Prøveundertype",
        },
        Object {
          "key": "status",
          "label": "Status",
        },
        Object {
          "key": "hasAnalyse",
          "label": "Analyse",
        },
        Object {
          "key": "showInfo",
          "label": "Vis detaljer",
        },
      ]
    }
    defaultSort={false}
    defaultSortDescending={false}
    filterBy=""
    hideFilterInput={false}
    itemsPerPage={0}
    noDataText="Ingen prøver funnet på objektet"
    sortBy={false}
    sortable={
      Array [
        "id",
        "doneDate",
      ]
    }
  >
    <Tr>
      <Td
        column="id"
      />
      <Td
        column="doneDate"
      >
        1991-11-11
      </Td>
      <Td
        column="sampleType"
      >
        Frø
      </Td>
      <Td
        column="sampleSubType"
      >
        Rogn
      </Td>
      <Td
        column="status"
      >
        2
      </Td>
      <Td
        column="hasAnalyse"
      />
      <Td
        column="showInfo"
      >
        <a
          href="en url"
        >
          Se mer / Rediger
        </a>
      </Td>
    </Tr>
    <Tr>
      <Td
        column="id"
      />
      <Td
        column="doneDate"
      >
        1991-11-12
      </Td>
      <Td
        column="sampleType"
      >
        Vev
      </Td>
      <Td
        column="sampleSubType"
      >
        Blod
      </Td>
      <Td
        column="status"
      >
        1
      </Td>
      <Td
        column="hasAnalyse"
      />
      <Td
        column="showInfo"
      >
        <a
          href="en url"
        >
          Se mer / Rediger
        </a>
      </Td>
    </Tr>
    <Tr>
      <Td
        column="id"
      />
      <Td
        column="doneDate"
      >
        2011-11-11
      </Td>
      <Td
        column="sampleType"
      >
        Tekstil
      </Td>
      <Td
        column="sampleSubType"
      >
        Bomull
      </Td>
      <Td
        column="status"
      >
        2
      </Td>
      <Td
        column="hasAnalyse"
      />
      <Td
        column="showInfo"
      >
        <a
          href="en url"
        >
          Se mer / Rediger
        </a>
      </Td>
    </Tr>
  </Table>
</div>
`;
